Neos.Flow/Classes/Http/Middleware/SessionMiddleware.php

```diff
@@ -37,30 +40,35 @@ class SessionRequestComponent implements ComponentInterface
      */
     protected $sessionManager;
 
-    /**
-     * @param ComponentContext $componentContext
-     */
-    public function handle(ComponentContext $componentContext)
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $next): ResponseInterface
     {
         if (!$this->sessionManager instanceof SessionManager) {
-            return;
+            return $next->handle($request);
         }
 
         $sessionCookieName = $this->sessionSettings['name'];
         /** @var ServerRequestInterface $request */
-        $request = $componentContext->getHttpRequest();
         $cookies = $request->getCookieParams();
 
         if (!isset($cookies[$sessionCookieName])) {
             $sessionCookie = $this->prepareCookie($sessionCookieName, Algorithms::generateRandomString(32));
             $this->sessionManager->createCurrentSessionFromCookie($sessionCookie);
-            return;
+            return $next->handle($request);
```

Comment:
We need to handle the setSession case here instead of return - that's the reason the test fails.